packages/babel-core/src/parse.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+
+import loadConfig, { InputOptions } from "./config";
+import normalizeFile from "./transformation/normalize-file";
+import normalizeOptions from "./transformation/normalize-opts";
+
+type AstRoot = BabelNodeFile | BabelNodeProgram;
+
+export default function parse(
+  code: string,
+  opts?: InputOptions,
+): AstRoot | null {
+  const config = loadConfig(opts);
+
+  if (config === null) {
+    return null;
+  }
+
+  const file = normalizeFile(config.passes, normalizeOptions(config), code);
```

Comment:
Is there any other logic in these that we'd want to avoid? I don't know off the top of my head.

Reply:
Good call - it might be better not to do all this extra stuff. My only question is if [this](https://github.com/babel/babel/blob/master/packages/babel-core/src/transformation/normalize-file.js#L64-L89) is needed. Other than that, I should be to just call `babylon.parse` directly with the normalized config's `parserOpts`.

Reply:
Hmm, I feel like that code probably should run so that we avoid differences between the `parse` and `transform`? Then we can aim for having `babel.parse()` basically be a nicer, faster version of `babel.transform` with no plugins?

Reply:
Sure, makes sense! This could always be optimized in the future.